java/arcs/sdk/wasm/WasmSingletonImpl.kt

```diff
@@ -34,19 +34,19 @@ class WasmSingletonImpl<T : WasmEntity>(
         }
     }
 
-    override fun onUpdate(action: (T?) -> Unit) {
+    override suspend fun onUpdate(action: (T?) -> Unit) {
         onUpdateActions.add(action)
     }
 
-    override fun fetch() = entity
+    override suspend fun fetch() = entity
 
-    override fun set(entity: T) {
+    override suspend fun set(entity: T) {
         this.entity = entity
         val encoded = entity.encodeEntity()
         WasmRuntimeClient.singletonSet(particle, this, encoded)
     }
 
-    override fun clear() {
+    override suspend fun clear() {
```

Comment:
Is anyone actually using WasmSingletonImpl? I have a feeling these change might work purely because no one is using them.